### PR TITLE
Enhancing integration test to show that "override" tags show up first

### PR DESCRIPTION
**Pull request by weaverryan**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | not needed

Relates a bit to #22396, in that I'm clarifying and emphasizing the following types of situations:

```yml
services:
    _instanceof:
        Symfony\Component\Security\Core\Authorization\Voter\VoterInterface:
            tags:
                # you probably shouldn't set priority here, but let's pretend we did
                - { name: security.voter, priority: 100 }

    AppBundle\Security\CoolPersonVoter:
        tags:
            - { name: security.voter, priority: 50 }
```

In the final `Definition`, the tags will appear in this order:
* security.voter, priority 50
* security.voter, priority 100

It works the same for parent-child definitions.

tl;dr; If a service has the same tag multiple times, the one that should be used should appear *earlier* in the Definition. The code already works that way - this test emphasizes it.

### CODE REVIEW

File: src/Symfony/Component/DependencyInjection/Tests/Compiler/IntegrationTest.php [hunk 1]
```diff
@@ -169,10 +169,10 @@ public function testInstanceofDefaultsAndParentDefinitionResolution()
         // tags inherit like normal
         $this->assertEquals(
             array(
-                'foo_tag' => array(array('priority' => 100), array()),
+                'foo_tag' => array(array('tag_option' => 'from_child_def'), array('tag_option' => 'from_parent_def'), array('tag_option' => 'from_instanceof')),
                 'bar_tag' => array(array()),
             ),
-            $simpleService->getTags()
+            $childDef2->getTags()
```
Comment (weaverryan) [inline src/Symfony/Component/DependencyInjection/Tests/Compiler/IntegrationTest.php]: This was actually a typo/bug that I made originally. Fixed now

**Comment by nicolas-grekas** [comment]
Thank you @weaverryan.
